### PR TITLE
change layernorm code to pytorch's native layer norm

### PR DESCRIPTION
**Pull request by dhpollack**
The current code basically recreates pytorch's native [LayerNorm](https://pytorch.org/docs/stable/nn.html#layernorm) code.  The only difference is that the default eps in the pytorch function is 1e-5 instead of 1e-12.  PyTorch's native version is optimized for cudnn so it should be faster than this version.

### CODE REVIEW

**Comment by honnibal** [comment]
I think your PR misses the point though? The models need to be 100% accurate reproductions of the Tensorflow code, right down to differences in eps values. Otherwise if you run the activations and get different results, you don't know whether there's a bug. You also can't reason about different results, and whether they matter.

So I think this PR needs to be closed.

**Comment by dhpollack** [comment]
@honnibal but looking at the code, every call of `BertLayerNorm` explicitly sets the eps, thus the actual values used in the BERT models does not change.  Only the default value, but this default value is never used.  Additionally, if APEX is available then you use `FusedLayerNorm`, which uses the [same default eps](https://github.com/NVIDIA/apex/blob/master/apex/normalization/fused_layer_norm.py#L70) of 1e-5 as the pytorch default `LayerNorm`.  So you already have an inconsistency, but you solved this by explicitly setting the eps every time you use the layer.

**Comment by honnibal** [comment]
Oh right! Fair point, sorry.

**Comment by thomwolf** [comment]
Yes @dhpollack is right we can switch to PyTorch official LayerNorm.

What made me reimplement the LayerNorm when I was working on Bert last year was actually a typo in PyTorch's doc formula for computing the LayerNorm which indicated, at that time, that the epsilon was added to the square root of the variance instead of being added to the variance it-self. This typo is now corrected in https://github.com/pytorch/pytorch/pull/8545.

Everything is right and we can drop these custom LayerNorms.

**Comment by julien-c** [comment]
Are we sure the names of the parameters are the same though? (`eps` vs. `variance_epsilon`)
